### PR TITLE
Bounty system Refactor

### PR DESCRIPTION
**Pull request by anjx123**
Heyy could you test my PR:

1. Delete your save file at:
C:/Users/[Your User]/AppData/LocalLow/S2dio/WastelandersGame

2. Validate that deck selection works fine after this data deletion:

3. See that levels are locked, go to your save file and play with your:
CurrentLevelProgress which should be at 0.0
value to verify that levels unlock properly. (A proper test would involve actually beating the level)

6. Verify that Contracts load the correct contract information
7. Verify that contract selection works properly, and that it handles edge cases like trying to select multiple bounties, and selecting no bounties
8. Press start on a bounty ensuring that it generally looks like the correct bounty has been selected. (Some bounties are incomplete for now)

9. Try to beat a bounty (Might be really hard I havn't balanced it yet), or play with the save file to see that bounties are correct when they have been beaten. And it shows up that its been beaten in the bounty seelct screen

### CODE REVIEW

**Comment by anjx123** [comment]
Reminder to rename Scene2 Scene file to the proper name now

File: Assets/Scripts/LevelSelect/BountySystem/PrincessFrogBounties.cs [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+            yield return DECREASED_HAND_SIZE;
+        }
+    }
+    public HashSet<IContracts> BountySet { get; }
+    public string FlavourText { get; }
+    public string BountyName { get; }
+
+    public string SceneName => GameStateManager.PRINCESS_FROG_BOUNTY;
+
+    public PrincessFrogBounties(HashSet<IContracts> bountySet, string challengeName, string flavourText)
```
Comment (j-ackyao) [inline Assets/Scripts/LevelSelect/BountySystem/PrincessFrogBounties.cs:68]: Should this be named contractSet instead of bountySet? assuming that bounties are a collection of contracts

File: Assets/Scripts/LevelSelect/BountySystem/BountySelect.cs [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+    }
+
+    public void StartLevel()
+    {
+        if (BountyManager.Instance.ActiveBounty != null)
+        {
+            OpenScene(BountyManager.Instance.ActiveBounty.SceneName);
+        } else
+        {
+            // TODO: Say select a bounty first !
```
Comment (j-ackyao) [inline Assets/Scripts/LevelSelect/BountySystem/BountySelect.cs:55]: Reminder of todo here

**Review by j-ackyao — APPROVED** [review]
LGTM, 7/7 tests pass
